volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -189,98 +223,64 @@ def get_module_info_for_module(
-
-    @staticmethod
-    def get_kernel_module_info(
-        context: interfaces.context.ContextInterface,
-        kernel_module_name: str,
-    ) -> ModuleInfo:
-        """
-        Returns a ModuleInfo instance that encodes the kernel
-        This is required to map function pointers to the kerenl executable
-        """
-        kernel = context.modules[kernel_module_name]
-
-        address_mask = context.layers[kernel.layer_name].address_mask
-
-        start_addr = kernel.object_from_symbol("_text")
-        start_addr = start_addr.vol.offset & address_mask
-
-        end_addr = kernel.object_from_symbol("_etext")
-        end_addr = end_addr.vol.offset & address_mask
-
-        return Modules.ModuleInfo(
-            start_addr, constants.linux.KERNEL_NAME, start_addr, end_addr
-        )
+        return ModuleInfo(module.vol.offset, mod_name, start, end)
 
     @classmethod
     def run_modules_scanners(
         cls,
         context: interfaces.context.ContextInterface,
-        kernel_name: str,
-        run_hidden_modules: bool = True,
+        kernel_module_name: str,
+        caller_wanted_gatherers: List[ModuleGathererInterface],
         flatten: bool = True,
-    ) -> Dict[str, List[ModuleInfo]]:
+    ) -> Dict[ModuleGathererInterface, List[ModuleInfo]]:
         """Run module scanning plugins and aggregate the results. It is designed
         to not operate any inter-plugin results triage.
 
+        Rules for `caller_wanted_sources`:
+            If `ModuleGatherers.all_gathers_identifier` is specified then every source will be populated
+
+            If empty or an invalid source is specified then a ValueError is thrown
         Args:
-            run_hidden_modules: specify if the hidden_modules plugin should be run
+            called_wanted_sources: The list of sources to gather modules.
+            flatten: Whether to de-duplicate modules across sources
         Returns:
             Dictionary mapping each plugin to its corresponding result
         """
+        if not caller_wanted_gatherers:
+            raise ValueError(
+                "`caller_wanted_gatherers` must have at least one gatherer."
+            )
 
-        kernel = context.modules[kernel_name]
+        if not isinstance(caller_wanted_gatherers, Iterable):
+            raise ValueError("`caller_wanted_gatherers` must be iterable")
 
-        address_mask = context.layers[kernel.layer_name].address_mask
+        for gatherer in caller_wanted_gatherers:
+            if not issubclass(gatherer, ModuleGathererInterface):
+                raise ValueError(
+                    f"Invalid gatherer sent through `caller_wanted_gatherers`: {gatherer}"
+                )
 
-        run_results = {}
+        kernel = context.modules[kernel_module_name]
 
-        # the kernel module boundaries
-        run_results["kernel"] = [cls.get_kernel_module_info(context, kernel_name)]
+        address_mask = context.layers[kernel.layer_name].address_mask
```

Comment:
I don't think this is needed here anymore?

Reply:
It is used in a call a few lines later.